{ed,gzip,gawk,gnugrep}: Add bash to buildInputs

rhelmot opened the pull request:
## Description of changes

part of https://github.com/NixOS/nixpkgs/pull/296581

These packages distribute shell scripts in /bin. Their shebangs are always patched to a /nix/store bash, but by default, they are patched to use the stdenv's shell, which during bootstrap is build in the previous stdenv. By using bash from the current stdenv, we can shorten the FreeBSD bootstrap process significantly.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] x86_64-freebsd
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

AndersonTorres commented:
You are modifying four different packages kept by different people.

Please, do one commit per package (no need for a new PR), so we can know what to bisect in a future bisection adventure.

rhelmot committed:
ed: Add runtimeShell to buildInputs

This package distributes shell scripts in /bin. Its shebangs are
always patched to a /nix/store shell, but by default, they are patched to
use the stdenv's shell, which during bootstrap is build in the previous
stdenv. By using runtimeShell from the current stdenv, we can shorten the
FreeBSD bootstrap process significantly.

rhelmot committed:
gzip: Add runtimeShell to buildInputs

This package distributes shell scripts in /bin. Its shebangs are
always patched to a /nix/store shell, but by default, they are patched to
use the stdenv's shell, which during bootstrap is build in the previous
stdenv. By using runtimeShell from the current stdenv, we can shorten the
FreeBSD bootstrap process significantly.

rhelmot committed:
gawk: Add runtimeShell to buildInputs

This package distributes shell scripts in /bin. Its shebangs are
always patched to a /nix/store shell, but by default, they are patched to
use the stdenv's shell, which during bootstrap is build in the previous
stdenv. By using runtimeShell from the current stdenv, we can shorten the
FreeBSD bootstrap process significantly.

rhelmot committed:
gnugrep: Add runtimeShell to buildInputs

This package distributes shell scripts in /bin. Its shebangs are
always patched to a /nix/store shell, but by default, they are patched to
use the stdenv's shell, which during bootstrap is build in the previous
stdenv. By using runtimeShell from the current stdenv, we can shorten the
FreeBSD bootstrap process significantly.

rhelmot commented:
I've split up the commit. Do you need me to take ownership of any of these packages?

AndersonTorres commented:
For now there is no need to keep maintainership of them. I will create a Nixpkgs team later this week to formalize this.

Ericson2314 reviewed (commented):
Does it make sense to use `runtimeShell` for this, or must these scripts use `bash` in particular?

rhelmot commented:
I didn't know about runtimeShell! I've changed it.
